implementation/contracts/deposit/DepositFunding.sol

```diff
@@ -103,6 +104,14 @@ library DepositFunding {
             block.timestamp > _d.signingGroupRequestedAt.add(TBTCConstants.getSigningGroupFormationTimeout()),
             "Signing group formation timeout not yet elapsed"
         );
+
```

Comment:
Are we validating that the group formation actually failed? We should probably check if the public key is set on the keep or something. If I understand correctly right now anyone can call `notifySignerSetupFailure` function even though the key is ready but `retrieveSignerPubkey` has not been called yet.

Reply:
We could add a check:
```
bytes memory _publicKey = IBondedECDSAKeep(_d.keepAddress).getPublicKey();
require(_publicKey != 64, "public key not set or set incorrectly");
```

Worth pointing out that this slightly changes calling incentives, removing any incentive for signers to call `retrieveSignerPubkey` since they can't be punished for not setting it within the timeout. 
@Shadowfiend Thoughts?

Reply:
> retrieveSignerPubkey

This method is one of my two least favorite parts of our protocol :rage4:

Reply:
> This method is one of my two least favorite parts of our protocol :rage4:

I hear you

Reply:
I'm 50-50 here. I think what we've got is likely sufficient, and am fine with a signer incentive to update this (the funder is also incentivized, since they'll be getting their funding back). This was one of the conclusions from the audit: even in imperfect cases, as long as players are properly incentivized, we're fine here.

I think this would be nice to have, but am wary of the incentive shift. But, I also don't love that it introduces a signer incentive that is a non-keep behavior other than rewards management. In particular, I don't like that fulfilling this incentive would require Go code that interacts with non-keep contracts.

Then again, I also despise that we have to actively pull the key from the keep to the deposit. I'm thinking we accept the imperfections and work to eliminate these bits in v2, though…

Reply:
I think we're on the same page here @Shadowfiend